integration/spark2/src/main/scala/org/apache/carbondata/store/SparkCarbonStore.scala

```diff
@@ -110,8 +110,16 @@ class SparkCarbonStore extends MetaCachedCarbonStore {
   }
 
   def startSearchMode(): Unit = {
+    LOG.info("Search mode master start service")
     master = new Master(session.sparkContext.getConf)
     master.startService()
+    var count = 0
+    while (!master.isStarted()) {
+      Thread.sleep(10)
```

Comment:
Is 10ms a proper period?
What if there is a situation that the master starts too late or failed and there will be a log explosion.

So better to add a maximum retrying counts or time and make the sleeping time dynamic by double it.

Reply:
ok